### ISSUE TITLE
False negatives on DLS_OVERWRITTEN_INCREMENT

### ISSUE THREAD

**Issue by gloNelson** [spotbugs/spotbugs#1517]
Hi, 
I found false negatives on `DLS_OVERWRITTEN_INCREMENT` when using SpotBugs to analyze our project.
Here is the demo,
```java
void test00(){
    int i = 0;
    i = 2 + (i++); // no warning (false negative)'
}

void test01(){
    int i = 0;
    i = (i++) + 2; // no warning (false negative)'
}

void test02(){
    int i = 0;
    i = (i--) - 2; // no warning (false negative)'
}


void test02(){
    int i = 0;
    i = 2 + (i--); // no warning (false negative)'
}
```

It seems that SpotBugs just simply supports the detection like `i = i++; ` or `i = i--`.
However, I think that based on its bug description, the pattern `DLS_OVERWRITTEN_INCREMENT` should provide more support to examples as I mentioned above.
